stl/inc/algorithm

```diff
@@ -2900,6 +2900,121 @@ _FwdIt3 transform(_ExPo&& _Exec, const _FwdIt1 _First1, const _FwdIt1 _Last1, co
+                                             _Get_unwrapped(_STD move(_First2)), _Get_unwrapped(_STD move(_Last2)),
+                                             _STD move(_Result), _Pass_fn(_Func), _Pass_fn(_Proj1), _Pass_fn(_Proj2));
+
+            _Seek_wrapped(_First1, _STD move(_Res.in1));
+            _Seek_wrapped(_First2, _STD move(_Res.in2));
+            return {_STD move(_First1), _STD move(_First2), _STD move(_Res.out)};
+        }
+
+        template <input_range _Rng1, input_range _Rng2, weakly_incrementable _Out, copy_constructible _Fn,
+            class _Pj1 = identity, class _Pj2 = identity>
+            requires indirectly_writable<_Out, indirect_result_t<_Fn&, projected<iterator_t<_Rng1>, _Pj1>,
+                                                                       projected<iterator_t<_Rng2>, _Pj2>>>
+        _NODISCARD constexpr binary_transform_result<borrowed_iterator_t<_Rng1>, borrowed_iterator_t<_Rng2>, _Out>
+            operator()(_Rng1&& _Range1, _Rng2&& _Range2, _Out _Result, _Fn _Func, _Pj1 _Proj1 = {}, _Pj2 _Proj2 = {})
+            const {
+            auto _Res = _Transform_unchecked(_Ubegin(_Range1), _Uend(_Range1),
+                                             _Ubegin(_Range2), _Uend(_Range2),
+                                             _STD move(_Result), _Pass_fn(_Func), _Pass_fn(_Proj1), _Pass_fn(_Proj2));
+
+            auto _First1 = _RANGES begin(_Range1);
+            auto _First2 = _RANGES begin(_Range2);
+            _Seek_wrapped(_First1, _STD move(_Res.in1));
+            _Seek_wrapped(_First2, _STD move(_Res.in2));
+            return {_STD move(_First1), _STD move(_First2), _STD move(_Res.out)};
+        }
+        // clang-format on
+    private:
+        template <class _It, class _Se, class _Out, class _Fn, class _Pj>
+        _NODISCARD constexpr unary_transform_result<_It, _Out> _Transform_unchecked(
+            _It _First, _Se _Last, _Out _Result, _Fn _Func, _Pj _Proj) const {
+            // transform projected [_First, _Last) with _Func
+            _STL_INTERNAL_STATIC_ASSERT(input_iterator<_It>);
+            _STL_INTERNAL_STATIC_ASSERT(weakly_incrementable<_Out>);
+            _STL_INTERNAL_STATIC_ASSERT(indirectly_writable<_Out, indirect_result_t<_Fn&, projected<_It, _Pj>>>);
+
+            for (; _First != _Last; ++_First, (void) ++_Result) {
+                *_Result = _STD invoke(_Func, _STD invoke(_Proj, *_First));
+            }
+
+            return {_STD move(_First), _STD move(_Result)};
+        }
+
+        template <class _It1, class _Se1, class _It2, class _Se2, class _Out, class _Fn, class _Pj1, class _Pj2>
+        _NODISCARD constexpr binary_transform_result<_It1, _It2, _Out> _Transform_unchecked(_It1 _First1, _Se1 _Last1,
+            _It2 _First2, _Se2 _Last2, _Out _Result, _Fn _Func, _Pj1 _Proj1, _Pj2 _Proj2) const {
+            // transform projected [_First1, _Last1) and projected [_First2, ...) with _Func

```

Comment:
This is a dual-range algorithm, not classic range-and-a-half:
```suggestion
            // transform projected [_First1, _Last1) and projected [_First2, _Last2) with _Func
```

Reply:
Done